text/0000-build-plan-merging.md

```diff
@@ -0,0 +1,260 @@
+# Meta
+[meta]: #meta
+
+- Name: Buildplan & Buildpack Plan Flags
+- Start Date: 2020-03-06
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The `metadata` field that is part of require should be removed and replaced with fields that have defined purpose to standardize inter-buildpack communication.
+
+# Motivation
+[motivation]: #motivation
+
+## State of Paketo Buildpacks
+- The current method that the Paketo Buildpacks uses to communicate information to an earlier buildpack is to set an arbitrary string field in the Build Plan (TOML) `requires.metadata` section.
+- Putting this information in the `requires.metadata` obfuscates the flow of information between buildpacks.
+- This implicit contract between dependent buildpacks should be formalized.
+
+### Example:
+ We consider the interactions between the Paketo `node-engine` and `npm`:
```

Comment:
This is really just for my understanding, but would it be possible to provide an example of what the build plan and buildpack plan look like with today's implementation? That might help to illustrate the changes introduced with this RFC.